source/linux/Foundational_Components/U-Boot/UG-Falcon-Mode.rst

```diff
@@ -0,0 +1,186 @@
+      => sf update $loadaddr 0x80000 $filesize
+
+eMMC Boot:
+----------
+
+In eMMC boot mode, the ``tiboot3.bin`` file should be flashed to the hardware
+boot partition whereas ``tifalcon.bin`` and the ``fitImage`` are read from
+the rootfs inside UDA. Use the U-Boot commands below to set the correct boot
+partition and write ``tiboot3.bin`` to the correct offset.
+
+.. code-block:: console
+
+   => # Set boot0 as the boot partition
+   => mmc partconf 0 1 1 1
+   => mmc bootbus 0 2 0 0
+   => # Flash tiboot3.bin to boot0
+   => mmc dev 0 1
+   => fatload mmc 1 ${loadaddr} tiboot3.bin
+   => mmc write ${loadaddr} 0x0 0x400
+
+For more information check: :ref:`How to flash eMMC and boot with eMMC Boot
+<how-to-emmc-boot>`.
+
+Custom fitImage creation:
+-------------------------
+
+Clone the `core-secdev-k3 source <https://git.ti.com/cgit/security-development-tools/core-secdev-k3>`__:
+
+.. code-block:: console
+
+   $ git clone https://git.ti.com/cgit/security-development-tools/core-secdev-k3
+
+Copy the required kernel image renamed to ``Image`` and the DTB renamed to
+``falcon.dtb`` inside the core-secdev-k3 source directory.
+
+Copy the following contents to a file named ``fitImage.its`` inside
+core-secdev-k3 source:
+
+.. code-block:: dts
+
+   /dts-v1/;
+
+   / {
+       description = "Kernel fitImage for falcon mode";
+       #address-cells = <1>;
+
+       images {
+           kernel-1 {
+               description = "Linux kernel";
+               data = /incbin/("Image.sec");
+               type = "kernel";
+               arch = "arm64";
+               os = "linux";
+               compression = "none";
+               load = <0x82000000>;
+               entry = <0x82000000>;
+           };
+           falcon.dtb {
+               description = "Flattened Device Tree blob";
+               data = /incbin/("falcon.dtb.sec");
+               type = "flat_dt";
+               arch = "arm64";
+               compression = "none";
+               load = <0x88000000>;
+           };
+       };
+
+       configurations {
+           default = "conf-falcon";
+           conf-falcon {
+               description = "Presigned Kernel and DTB";
+               kernel = "kernel-1";
+               fdt = "falcon.dtb";
+           };
+       };
+   };
+
+Sign the kernel and dtb using ``secure-binary-image.sh`` and create the
+``fitImage`` using mkimage:
+
+.. code-block:: console
+
+   $ # inside core-secdev-k3 source
+   $ ./scripts/secure-binary-image.sh Image Image.sec
+   $ ./scripts/secure-binary-image.sh falcon.dtb falcon.dtb.sec
+   $ mkimage -f fitImage.its fitImage
+
+----------------------
+Boot time comparisons:
+----------------------
+
+Removing A-core SPL and U-Boot from the boot process leads to ~60% reduction in
+time to kernel. Saving about 1-2 seconds during boot depending on the platform.
+
+**Falcon Mode (Left) vs Regular Boot (Right):**
+
+  .. Image:: /images/U-Boot_Falcon_Comparison.gif
```

Comment:
This sequence can be replaced with a [`figure`](https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure)